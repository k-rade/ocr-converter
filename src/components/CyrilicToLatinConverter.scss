.cyrilic-to-latin-converter {
    display: flex;
    flex-direction: column;
    gap: 1.5rem;

    &-textarea {
        padding: 1rem;
        border-radius: 2px;
        resize: none;
        background-color: var(--color-gray-200);
        border: 2px solid var(--color-gray-700);
        color: var(--color-white-one);
        outline: none;
        width: 100%;
        caret-color: var(--color-primary-500);
        font-family: var(--font);
        font-size: 0.875rem;
        font-weight: 400;

        &-label {
            display: block;
            font-weight: 500;
            color: var(--color-gray-900);
            margin-bottom: 0.5rem;
        }
    }
}