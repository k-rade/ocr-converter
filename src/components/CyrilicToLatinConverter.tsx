import { ChangeEvent, useState } from "react";
import { convertCyrilicToLatin } from "../util/CyrilicToLatinConveterUtils";
import TextWrapper from "./TextWrapper";
import Breadcrumbs from "./Breadcrumbs";

import "./CyrilicToLatinConverter.scss";

const CyrilicToLatinConverter = (): JSX.Element => {
  const [text, setText] = useState("");

  const handleTextChange = (e: ChangeEvent<HTMLTextAreaElement>): void => {
    setText(e.target.value);
  };
  return (
    <div className="cyrilic-to-latin-converter">
      <Breadcrumbs />
      <div>
        <label
          htmlFor="cyrilic-to-latin-converter-textarea"
          className="cyrilic-to-latin-converter-textarea-label"
        >
          Input
        </label>
        <textarea
          id="cyrilic-to-latin-converter-textarea"
          className="cyrilic-to-latin-converter-textarea"
          value={text}
          onChange={handleTextChange}
          rows={10}
          placeholder="Enter cyrilic text here"
        />
      </div>
      <TextWrapper text={convertCyrilicToLatin(text)} />
    </div>
  );
};

export default CyrilicToLatinConverter;
