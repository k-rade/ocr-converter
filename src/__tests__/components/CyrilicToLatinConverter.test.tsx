import { fireEvent, render } from "@testing-library/react";
import CyrilicToLatinConverter from "../../components/CyrilicToLatinConverter";
import { MemoryRouter, useLocation } from "react-router-dom";

jest.mock("react-router-dom", (): any => {
  const actualImplementation = jest.requireActual("react-router-dom");
  return {
    ...actualImplementation,
    useLocation: jest.fn(),
  };
});

describe("CyrilicToLatinConverter", (): void => {
  const TestComponent = (
    <MemoryRouter>
      <CyrilicToLatinConverter />
    </MemoryRouter>
  );

  beforeEach((): void => {
    (useLocation as jest.Mock).mockImplementation(() => ({
      pathname: "/test-route",
    }));
  });

  it("should render", (): void => {
    const { baseElement } = render(TestComponent);

    expect(baseElement).toMatchSnapshot();
  });

  it("should convert text from cyrilic to latin letters", (): void => {
    const { getByLabelText, getByText } = render(TestComponent);

    fireEvent.change(getByLabelText("Input"), {
      target: {
        value:
          "Ћирилица је писмо које се користи у неколико словенских језика, као што су: белоруски, бугарски, македонски, руски, русински, српски и украјински",
      },
    });

    expect(
      getByText(
        "Ćirilica je pismo koje se koristi u nekoliko slovenskih jezika, kao što su: beloruski, bugarski, makedonski, ruski, rusinski, srpski i ukrajinski"
      )
    ).toBeInTheDocument();
  });
});
