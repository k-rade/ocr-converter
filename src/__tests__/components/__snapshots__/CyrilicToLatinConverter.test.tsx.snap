// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CyrilicToLatinConverter should render 1`] = `
<body>
  <div>
    <div
      class="cyrilic-to-latin-converter"
    >
      <div
        class="breadcrumbs"
      >
        <a
          aria-current="page"
          class="breadcrumb breadcrumb-active"
          href="/"
        >
          Home
        </a>
        <span
          class="breadcrumb-divider"
        >
          &gt;
        </span>
        <a
          class="breadcrumb "
          href="/test-route"
        >
          test route
        </a>
      </div>
      <div>
        <label
          class="cyrilic-to-latin-converter-textarea-label"
          for="cyrilic-to-latin-converter-textarea"
        >
          Input
        </label>
        <textarea
          class="cyrilic-to-latin-converter-textarea"
          id="cyrilic-to-latin-converter-textarea"
          placeholder="Enter cyrilic text here"
          rows="10"
        />
      </div>
      <div
        class="text-wrapper"
      >
        <div
          class="text-wrapper-heading"
        >
          Output
        </div>
        <textarea
          class="text-wrapper-output"
          disabled=""
          placeholder="Output text will appear here"
          rows="10"
        />
        <div
          class="text-wrapper-actions"
        >
          <div
            class="icon-button-container"
          >
            <span
              class="icon-button-tooltip"
            >
              Copy text
            </span>
            <button
              class="icon-button"
              disabled=""
              type="button"
            >
              <svg>
                copy.svg
              </svg>
            </button>
          </div>
          <div
            class="icon-button-container"
          >
            <span
              class="icon-button-tooltip"
            >
              Download
            </span>
            <button
              class="icon-button"
              disabled=""
              type="button"
            >
              <svg>
                download-file.svg
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
